### DISCUSSION TITLE
Zigbee Thermostat

### DISCUSSION THREAD

**Discussion by piokacz** [milo526/homebridge-tuya-web#155] [Q&A] [answered]
Hi,
I'm using zigbee Thermostat. I can see it in Homebridge. My configuration bellow
                {
                    "id": "bfe33e910004e2ba8aa8i1",
                    "device_type": "climate",
                    "min_temper": "16",
                    "max_temper": "26",
                    "current_temperature_factor": "0.1",
                    "target_temperature_factor": "0.1"
                }
But I can not force it to work in home bridge - "device not responding"
Home bridge log:
[TuyaWebPlatform] [Termostat salon] - [SET] Invalid payload in response: {"payload":{},"header":{"payloadVersion":1,"code":"ValueOutOfRange"}}
Of course I tried to change both factors to 1 and nothing.

**Comment by milo526** [comment, marked as answer]
`ValueOutOfRange` means that the value that homekit is providing is bigger than the maximum value that your thermostat supports.

You will probably need to play with your `target_temperature_factor` and `max_temper` to get a good combination.

Please post the result of [the steps under "How to check whether the api this library uses can control your device".](https://github.com/milo526/homebridge-tuya-web#how-to-check-whether-the-api-this-library-uses-can-control-your-device)

Please post some more log output.

With the both of those I can look into this some more.

**Reply by piokacz** [reply]
Hi Milo, many thanks for Your answer.

In TuyaSmart app I can set the minimum temp to 5 degrees and the max is 35 - but in HomeKit I set 16 and 26 as maximum - changing it to 5 and 35 did nothing.

There is a real.it report with my thermostat:

Getting credentials
Got credentials
{   'access_token': 'EUheu16________75X5bSKFqarzMfrxQ',
    'expires_in': 864000,
    'refresh_token': 'EUheu16________75X5bSKPRj9EDHp69',
    'token_type': 'bearer'}
Getting devices
Got devices
{   'header': {'code': 'SUCCESS', 'payloadVersion': 1},
    'payload': {   'devices': [   {   

(….)

                                  {   'data': {   'current_temperature': 220,
                                                  'max_temper': 350,
                                                  'min_temper': 50,
                                                  'online': True,
                                                  'state': 'true',
                                                  'temperature': 220},
                                      'dev_type': 'climate',
                                      'ha_type': 'climate',
                                      'icon': 'https://images.tuyaeu.com/smart/icon/ay15392________71a8dbbfcbcbd.png',
                                      'id': 'bfe33________a8aa8i1',
                                      'name': 'Termostat salon'},

                                (….)

!!! NOW REMOVE THIS FILE, SO YOUR CREDENTIALS (username, password) WON'T LEAK !!!

Below my config.json

                {
                    "id": "bfe33e9_______a8aa8i1",
                    "device_type": "climate",
                    "min_temper": "16",
                    "max_temper": "26",
                    "current_temperature_factor": "0.1",
                    "target_temperature_factor":  "0.1"
                }

Only if I set "current_temperature_factor" to 0.1 the thermostat shows real temperature in the room.

I tried to set "target_temperature_factor" to 0.01 ; 0.1 ; 0.5 ; 1 ; 2 - but I always get this in the log:

[1/13/2021, 10:37:13 PM] [TuyaWebPlatform] [Termostat salon] - [SET] Invalid payload in response: {"payload":{},"header":{"payloadVersion":1,"code":"ValueOutOfRange"}}

I can switch on and off my thermostat, but I can not change the temperature.

Below more from homebridge status:

[1/13/2021, 11:03:17 PM] [TuyaWebPlatform] [Termostat salon] - Renewing cache due to RateLimitError
[1/13/2021, 11:03:17 PM] [TuyaWebPlatform] [Termostat salon] Characteristic.CurrentTemperature - [GET] 220
[1/13/2021, 11:03:17 PM] [TuyaWebPlatform] [Termostat salon] Characteristic.CurrentTemperature - [UPDATE] 22
[1/13/2021, 11:03:17 PM] [TuyaWebPlatform] [Termostat salon] Characteristic.TargetTemperature - [GET] 2100
[1/13/2021, 11:03:17 PM] [TuyaWebPlatform] [Termostat salon] Characteristic.TargetTemperature - [UPDATE] 21
[1/13/2021, 11:03:17 PM] [TuyaWebPlatform] [Termostat salon] Characteristic.CurrentHeatingCoolingState - [GET] { state: 'true', mode: undefined }
[1/13/2021, 11:03:17 PM] [TuyaWebPlatform] [Termostat salon] Characteristic.CurrentHeatingCoolingState - [UPDATE] HEAT
[1/13/2021, 11:03:17 PM] [TuyaWebPlatform] [Termostat salon] Characteristic.TargetHeatingCoolingState - [GET] { state: 'true', mode: undefined }
[1/13/2021, 11:03:17 PM] [TuyaWebPlatform] [Termostat salon] Characteristic.TargetHeatingCoolingState - [UPDATE] AUTO

**Reply by milo526** [reply]
I'm afraid you will have to experiment some more sadly.

Tuya does not give any way to identify the valid range for a device. Your best hope is just to extend the values and try some more.
I know this is not a useful reply but it is the best I got 😢
